sql/core/src/test/scala/org/apache/spark/sql/DataFrameTimeWindowingSuite.scala

```diff
@@ -575,4 +575,61 @@ class DataFrameTimeWindowingSuite extends QueryTest with SharedSparkSession {
       validateWindowColumnInSchema(schema2, "window")
     }
   }
+
+  test("window_time function on raw window column") {
+    val df = Seq(
+      ("2016-03-27 19:38:18"), ("2016-03-27 19:39:25")
+    ).toDF("time")
+
+    checkAnswer(
+      df.select(window($"time", "10 seconds").as("window"))
+        .select(
+          $"window.end".cast("string"),
+          window_time($"window").cast("string")
+        ),
+      Seq(
+        Row("2016-03-27 19:38:20", "2016-03-27 19:38:19.999999"),
+        Row("2016-03-27 19:39:30", "2016-03-27 19:39:29.999999")
+      )
+    )
+  }
+
+  test("2 window_time functions on raw window column") {
+    val df = Seq(
+      ("2016-03-27 19:38:18"), ("2016-03-27 19:39:25")
+    ).toDF("time")
+
+    checkAnswer(
+      df.select(window($"time", "10 seconds").as("window"))
+        .select(
+          $"window.end".cast("string"),
+          window_time($"window").cast("string").as("window1"),
```

Comment:
This is very interesting... @cloud-fan Any idea on this? Do we capture the expression as the same? Otherwise it sounds to be a be odd to pass the condition below:

```
val child = p.children.head
val windowTimeExpressions =
  p.expressions.flatMap(_.collect { case w: WindowTime => w }).toSet

if (windowTimeExpressions.size == 1
```

Reply:
OK this was pretty straightforward... Refer the definition of WindowTime.

`case class WindowTime(windowColumn: Expression)`

We apply `.toSet` to collect the instances with deduplication, so as long as the windowColumn is resolved as the same, they are considered to the same.

Reply:
Thanks for figuring this out. Redid the test

Reply:
Although the test is failing now, it's failing due to rule of time window and corresponding unsupported ops checker. Is it intentional? I guess we want to test the case of multiple window_time.

Refer this comment: https://github.com/apache/spark/pull/38288#discussion_r1002572937